index.js

```diff
@@ -18,20 +18,41 @@ module.exports = function (cmd, args, opts) {
 				val = stripEof(val);
 			}
 
+			if (parsed.options.encoding !== undefined) {
+				val = new Buffer(val, parsed.options.encoding);
+			}
+
 			return val;
 		};
 
-		var spawned = childProcess.execFile(parsed.command, parsed.args, parsed.options, function (err, stdout, stderr) {
-			if (err) {
-				err.stdout = stdout;
-				err.stderr = stderr;
-				reject(err);
-				return;
-			}
+		var spawned = childProcess.spawn(parsed.command, parsed.args, parsed.options);
+		var out = '';
+		var err = '';
+		var all = '';
+
+		spawned.stdout.on('data', function (data) {
+			out += data.toString();
```

Comment:
Using `.toString()` is a common mistake in Node.js, but this will totally break on unicode where one character might be split between different `data` chunks. Instead use `.setEncoding()` upfront.